_posts/2021/2021-12-06-tidy-up-your-routes-with-only.md

```diff
@@ -0,0 +1,53 @@
+---
+title: "Tidy Up Your Routes with Only"
+description: "Tidy routes are happy routes"
+layout: article
+category: ruby
+image:
+  base: "2021/tidy-up-your-routes-with-only"
+  alt: "Road Closed. Diversion. Road signs."
+  credit: "Ben Wicks"
+  source: "https://unsplash.com/photos/zwN1MwCtR5Y"
+
+---
+
+It is easy with [Rails’ syntax for defining routes](https://guides.rubyonrails.org/routing.html) to open up more URLs in your application than you might intend.
+
+
+## Instead of…
+
+…using an open `resources` block in your routes:
+
+```ruby
+resources :orders do
+  resources :products
+end
+```
+
+## Use…
+
+…the `only` and `except` options to limit the actions you’re generating:
+
+```ruby
+resources :orders, except: %w[destroy] do
+  resources :products, only: %w[show]
+end
+```
+
+
+## Why?
+
+This is all about clarity, tidying up and protecting against unexpected errors or security holes.
+
+Primarily the issue is that in the default, unrestricted, case all seven default routes are created. If a user calls a route that is defined but not used, Rails will attempt to call the relevent controller action, even if it doesn’t exist. This can lead to errors or, worse, unexpected behaviour.
```

Comment:
*In the default, unrestricted case, seven routes are created. If a user calls a route...

Reply:
```suggestion
In the default, unrestricted, case all seven default routes are created. If a user calls a route that is defined but not used, Rails will attempt to call the relevent controller action, even if it doesn’t exist. This can lead to errors or, worse, unexpected behaviour.
```